src/dpo/src/detailed_manager.cxx

```diff
@@ -1384,6 +1384,84 @@ int DetailedMgr::checkEdgeSpacingInSegments()
   return err_n + err_p;
 }
 
+std::vector<std::vector<int>> DetailedMgr::getOneSiteGapViolationsPerSegment()
+{
+  // the pair is the segment and the cell index in that segment
+  std::vector<std::vector<int>> violating_cells;
+  violating_cells.resize(segments_.size() + 1);
+
+  std::vector<Node*> temp;
+  temp.reserve(network_->getNumNodes());
+  for (int s = 0; s < segments_.size(); s++) {
+    // To be safe, gather cells in each segment and re-sort them.
+    temp.clear();
+    for (Node* ndj : cellsInSeg_[s]) {
+      temp.push_back(ndj);
+    }
+    std::sort(temp.begin(), temp.end(), compareNodesX());
+
+    // The idea here is to get the last X before the current cell,
+    // it doesn't have to be the one directly before the current cell as
+    // there might be two cells on the same x but different y.
+    // if the difference between the last X and the current cell's X is equal to
+    // 1-site then we might have a violation. we still need to check the
+    // overlaps in the ys
+    // to do this efficiently we can simply loop and check for overlaps in the
+    // ys
+
+    auto isInRange = [](int value, int min, int max) -> bool {
+      return value >= min && value <= max;
```

Comment:
purely stylistic but I prefer 

min <= value  && value <= max

as it reads more like the mathematical

min <= value <= max